scripts/common/primer_autodetection.py

```diff
@@ -66,11 +65,11 @@ def build_primers_automaton(primer_index: Path) -> dict[str, PrimerAutomaton]:
                 TOTAL_NUM_PRIMER: int(row[TOTAL_NUM_PRIMER]),
                 PRIMER_AUTODETECTION_NUMREADS_COL: 0,
                 PRIMER_AUTODETECTION_UNIQUE_NUMREADS_COL: 0,
-                PRIMER_AUTODETECTION_AMBIGUOUS_NUMREADS_COL: 0,
             }
 
             # this can be optimised by generating the automaton at build time (one-off)
             automaton = load_pickle(primer_pickle_path)
+            automaton.make_automaton()
```

Comment:
This script also contains critical changes.

This is now necessary as we no longer search from the beginning of the read.